validator/sawtooth_validator/gossip/structure_verifier.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import logging
+# pylint: disable=import-error,no-name-in-module
+# needed for google.protobuf import
+from google.protobuf.message import DecodeError
+
+from sawtooth_validator.protobuf import client_pb2
+from sawtooth_validator.protobuf.batch_pb2 import BatchHeader
+from sawtooth_validator.protobuf.batch_pb2 import Batch
+from sawtooth_validator.protobuf.block_pb2 import Block
+from sawtooth_validator.protobuf import network_pb2
+from sawtooth_validator.protobuf.network_pb2 import GossipBlockResponse
+from sawtooth_validator.protobuf.network_pb2 import GossipBatchResponse
+from sawtooth_validator.networking.dispatch import Handler
+from sawtooth_validator.networking.dispatch import HandlerResult
+from sawtooth_validator.networking.dispatch import HandlerStatus
+from sawtooth_validator.protobuf.validator_pb2 import Message
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def is_valid_block(block):
+    if not all(map(is_valid_batch, block.batches)):
+        return False
+
+    return True
```

Comment:
I personally think that this is cleaner:
`return all(map(is_valid_batch, block.batches))` or as @delventhalz recommended:
`return all(is_valid_batch(batch) for batch in block.batches)` but won't let that hold up the review.